### ISSUE TITLE
[SIIT] Convert MPT to support gradient checkpointing, LoRa

### ISSUE THREAD

**Issue by ArthurMinovsky** [OpenThaiGPT/openthaigpt-pretraining#180]
Ref: https://github.com/iwalton3/mpt-lora-patch

- [ ] Gradient Checkpoint
- [ ] LoRa

**Comment by Wintoplay** [comment]
new repository: https://github.com/mikeybellissimo/LoRA-MPT/tree/main

https://colab.research.google.com/drive/1rPwsy659VxOMyx7yUiYo6tsfsg_SyAuf?usp=sharing

still fail: loss == 0

**Comment by Chawak** [comment]
Resource 
- https://github.com/mikeybellissimo/LoRA-MPT/issues/6
- https://colab.research.google.com/drive/1yQZdMdZlMwpniF2cQlTAPnNWqZSDmmb1?usp=sharing

**Comment by Wintoplay** [comment]
https://colab.research.google.com/drive/1qdRMaJ-ro2k7k23d-ICoBq4IjSwiaTnA?usp=sharing

**Comment by Wintoplay** [comment]
@pitikorn32 (@boss-chanon บอกให้ tag)
repo: https://github.com/mikeybellissimo/LoRA-MPT/tree/main

error message after upgrading peft lib:
/tmp/tmpfgiig51b/main.c:2:10: fatal error: cuda.h: No such file or directory
 #include "cuda.h"
